connect/transforms/src/main/java/org/apache/kafka/connect/transforms/TimestampConverter.java

```diff
@@ -375,15 +389,15 @@ private Struct applyValueWithSchema(Struct value, Schema updatedSchema) {
     }
 
     private R applySchemaless(R record) {
-        if (config.field.isEmpty()) {
-            Object value = operatingValue(record);
-            return newRecord(record, null, convertTimestamp(value));
-        } else {
-            final Map<String, Object> value = requireMap(operatingValue(record), PURPOSE);
-            final HashMap<String, Object> updatedValue = new HashMap<>(value);
-            updatedValue.put(config.field, convertTimestamp(value.get(config.field)));
-            return newRecord(record, null, updatedValue);
+        Object rawValue = operatingValue(record);
+        if (rawValue == null || config.field.isEmpty()) {
+            Object value = convertTimestamp(rawValue);
+            return newRecord(record, null, value);
         }
+        final Map<String, Object> value = requireMap(rawValue, PURPOSE);
+        final HashMap<String, Object> updatedValue = new HashMap<>(value);
+        updatedValue.put(config.field, convertTimestamp(value.get(config.field)));
+        return newRecord(record, null, updatedValue);
```

Comment:
I understand that the if block always returns, and so an `else` block is unnecessary. But because we likely want to backport this, and because we'd like to minimize the changes to help ensure the behavior remains the same for non-null timestamps, it's probably worth it to not remove the `else` block and keep the original indentation.